src/main/java/com/google/devtools/build/lib/actions/ResourceManager.java

```diff
@@ -392,6 +406,21 @@ private boolean areResourcesAvailable(ResourceSet resources) {
     double availableIo = availableResources.getIoUsage();
     int availableLocalTestCount = availableResources.getLocalTestCount();
 
+    double remainingRam = availableRam - usedRam;
+
+    if (localMemoryEstimate) {
+      try {
+        ProcMeminfoParser memInfo = new ProcMeminfoParser();
+        double totalFreeRam = memInfo.getFreeRamKb() / 1024;
+        double reserveMemory = staticResources.getMemoryMb() * (100.0 - this.ramUtilizationPercentage) / 100.0;
+        remainingRam = totalFreeRam - reserveMemory;
+      } catch (IOException e) {
+        // Ignore
```

Comment:
What should happen if this fails?

Reply:
Well, right now, we fall back to leaving `remainingRam` at a value that preserves the legacy behavior.  Do you think I should

a) Add a better comment describing how that happens
b) Add some form of logging?  I'm not super familiar with the bazel code-base, but you wouldn't want to log it every time, maybe only on the first failure.
c) Report some error and halt?

Reply:
Let's start with a clear comment that it's intentionally falling back to the old code. I don't immediately see any way to log at this point in the code. ResourceManager doesn't do any logging right now, and I'm reluctant to add any 'just' for this.

Conceptually, when can these exceptions be thrown, and what do you think the right behavior should be in that case?

Reply:
Comment updated, including reasoning about what can cause those exceptions to be thrown.